### PR TITLE
Update gen_distribution_version_testcase.py

### PR DESCRIPTION
**Pull request by gargtanishka895**
chnges i made will come up with optimizations that will reduce redundant file checks and improve code readability without affecting the functionality of the script.

##### ISSUE TYPE

- Bugfix Pull Request
- Docs Pull Request
- Feature Pull Request
- Test Pull Request

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: hacking/tests/gen_distribution_version_testcase.py [hunk 1]
```diff
@@ -41,39 +41,28 @@
...
+]))
 
-for f in filelist:
-    if os.path.exists(f):
-        s = os.path.getsize(f)
-        if s > 0 and s < 10000:
-            with open(f) as fh:
-                fcont[f] = fh.read()
+# Use list comprehension to filter and read file contents
+fcont = {f: open(f).read() for f in filelist if os.path.exists(f) and 0 < os.path.getsize(f) < 10000}
```
Comment (KalleOlaviNiemitalo) [inline hacking/tests/gen_distribution_version_testcase.py:47]: This leaves the files open, doesn't it?  In the original code, `with open(f) as fh:` closed each file.

Comment (Akasurde) [reply hacking/tests/gen_distribution_version_testcase.py:47]: Original code looks good to me.

File: hacking/tests/gen_distribution_version_testcase.py [hunk 2]
```diff
@@ -41,39 +41,28 @@
...
 
 try:
-    b_ansible_out = subprocess.check_output(
-        ['ansible', 'localhost', '-m', 'setup'])
+    b_ansible_out = subprocess.check_output(['ansible', 'localhost', '-m', 'setup'])
+    ansible_out = to_text(b_ansible_out)
+    parsed = json.loads(ansible_out[ansible_out.index('{'):])
+    ansible_facts = {}
+    for fact in facts:
+        ansible_facts[fact] = parsed['ansible_facts'].get('ansible_' + fact, "N/A")
```
Comment (Akasurde) [inline hacking/tests/gen_distribution_version_testcase.py:59]: None of these lines will raise `CalledProcessError` exception so there is no point in including these lines in the `try-except` block.

**Comment by Akasurde** [comment]
Closing due to inactivity.

**Pull request closed without merging**